### PR TITLE
[AI-FSSDK] [FSSDK-12369] Add local holdouts support to JavaScript SDK

### PR DESCRIPTION
**Pull request by Mat001**
## Summary

Adds Local Holdouts support to the JavaScript SDK. Local holdouts allow targeting specific experiment or delivery rules within a flag, while global holdouts (the existing behavior) apply to all rules across all flags. This introduces a new rule-level targeting system using an `includedRules` field on the holdout data model.

## Changes

- Added `includedRules` and `isGlobal` fields to the `Holdout` interface in `shared_types.ts`
- Added `HoldoutConfig` interface and `getGlobalHoldouts` / `getHoldoutsForRule` helper functions to `project_config.ts`; `parseHoldoutsConfig` now builds and stores these lookups during config initialization
- Updated `decision_service/index.ts` to evaluate global holdouts via `getGlobalHoldouts()` at flag level and check local holdouts per-rule (after forced decisions, before regular rule evaluation) for both experiment rules and delivery rules
- Added Level 1 unit tests for config/parsing (isGlobal classification, backward compat, edge cases) and Level 2 decision service tests (one per branch of the local holdout pseudocode)

## Jira Ticket

[FSSDK-12369](https://optimizely-ext.atlassian.net/browse/FSSDK-12369)

[FSSDK-12369]: https://optimizely-ext.atlassian.net/browse/FSSDK-12369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/core/decision_service/index.ts [hunk 1]
```diff
@@ -145,6 +149,12 @@ interface UserProfileTracker {
 type VarationKeyWithCmabParams = {
   variationKey?: string;
   cmabUuid?: string;
+  /**
+   * Set when the variation comes from a local holdout (FSSDK-12369).
+   * When present, the caller should use this as the full decision result
+   * instead of constructing one from the experiment.
+   */
+  localHoldoutDecision?: DecisionObj;
```
Comment (raju-opti) [inline lib/core/decision_service/index.ts]: This field does not make sense inside this type. This will work functionally now, but I would like to refactor it later

File: lib/core/decision_service/index.spec.ts [hunk 2]
```diff
@@ -2879,4 +2879,227 @@ describe('DecisionService', () => {
       expect(variation).toBe(null);
     });
   });
+
+  // Level 2 decision service tests for local holdouts (FSSDK-12369)
+  // One test per branch of the pseudocode in Step 3 of the ticket.
+  describe('local holdouts (FSSDK-12369)', () => {
```
Comment (raju-opti) [inline lib/core/decision_service/index.spec.ts]: please remove ticket number from the title

File: lib/core/decision_service/index.spec.ts [hunk 3]
```diff
@@ -2879,4 +2879,227 @@ describe('DecisionService', () => {
       expect(variation).toBe(null);
     });
   });
+
+  // Level 2 decision service tests for local holdouts (FSSDK-12369)
```
Comment (raju-opti) [inline lib/core/decision_service/index.spec.ts]: please remove the ticket number

**Review by raju-opti — APPROVED** [review]
I would like to refactor few thing. But good for now
